main.tf

```diff
@@ -0,0 +1,90 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 2.49"
+    }
+  }
+
+  backend "azurerm" {
+    resource_group_name  = "SoftwirePilot_JoannaCheng_ProjectExercise"
+    storage_account_name = "joachetstate"
+    container_name       = "joachetstate"
+    key                  = "terraform.tfstate"
+  }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_resource_group" "main" {
+  name = var.resource_group_name
+}
+
+resource "azurerm_app_service_plan" "main" {
+  name                = "terraformed-asp"
+  location            = var.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  kind                = "Linux"
+  reserved            = true
+
+  sku {
+    tier = "Basic"
+    size = "B1"
+  }
+}
+
+resource "azurerm_app_service" "main" {
+  name                = "terraformed-todo-app-joache"
```

Comment:
Not vital here, but often nice to provide a pre/post-fix for these names, so that we can easily spin up multiple environments